src/core/server/saved_objects/import/check_conflict_for_data_source.ts

```diff
@@ -74,12 +92,37 @@ export async function checkConflictsForDataSource({
         /**
          * Only update importIdMap and filtered objects
          */
+
+        // Some visualization types will need special modifications, like Vega visualizations
+        if (object.type === 'visualization') {
+          const vegaSpec = extractVegaSpecFromSavedObject(object);
+
+          if (!!vegaSpec) {
+            const previousDataSourceTitle =
+              previoudDataSourceId && savedObjectsClient
+                ? await getDataSourceTitleFromId(previoudDataSourceId, savedObjectsClient)
+                : undefined;
```

Comment:
Can we avoid making individual calls if we can make a bulk find request?

Reply:
Removed repeated individual `get` requests